cell_classification/segmentation_data_prep.py

```diff
@@ -182,19 +188,53 @@ def prepare_example(self, data_folder, marker):
             "marker_activity_mask": marker_activity_mask.astype(np.uint8),
             "dataset": self.dataset,
             "marker": marker,
-            "cell_types": cell_types,
+            "cell_types": marker_activity,
+            "folder_name": fov,
         }
 
-    def tile_example(example, tile_size):
+    def tile_example(
+        self, example, spatial_keys=[
+            "mplex_img", "binary_mask", "instance_mask", "marker_activity_mask",
+        ],
+    ):
         """Tiles the example into a grid of tiles
         Args:
             example (dict):
                 The example to tile
+            tile_size (list):
+                The size of the tiles
+            stride (list):
+                The stride of the tiles
+            spatial_keys (list):
+                The keys in the example to tile
         Returns:
             list:
                 List of example dicts, one for each tile
         """
-        return None
+        # tile the example
+        tiled_examples = {}
+        for key in spatial_keys:
+            if example[key].ndim == 2:
+                example[key] = np.expand_dims(example[key], axis=-1)
+            res = np.lib.stride_tricks.sliding_window_view(
+                example[key], window_shape=self.tile_size + list(example[key].shape[2:])
+            )[:: self.stride[0], :: self.stride[1]]
+            sh = list(res.shape)
+            tiled_examples[key] = res.reshape([sh[0] * sh[1]] + sh[3:])  # tiles x H x W x C
+
+        # store individual tiled examples in a list of dicts
+        non_spatial_keys = [key for key in example if key not in spatial_keys]
```

Comment:
This is fine for now. We should add an issue though that the non-spatial metadata for each tile will not have been subsetted to include only cells in the tile, but all cells from the original untiled image. If we decide to use these non-spatial keys for assessing accuracy, we'll need to either subset then, or decide to change this part and do subsetting here, depending on which makes more sense

Reply:
That's a good idea. I just implemented it since I don't really see a downside. If we see a problem with this in the feature I'll make it optional.